### PR TITLE
Move to using GraphQL mutations for SdmGoal and SdmGoalSet

### PR DESCRIPTION
**Pull request by cdupuis**
This relates to atomist/sdm#788

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -84,19 +84,6 @@ export {
...
-    SpawnedDeployment,
-    StartupInfo,
-} from "./lib/internal/delivery/deploy/local/LocalDeployerOptions";
-export {
-    DeployedApp,
-    LookupStrategy,
-    ManagedDeployments,
-    ManagedDeploymentTargeter,
-    ManagedDeploymentTargetInfo,
-} from "./lib/internal/delivery/deploy/local/ManagedDeployments";
```
Comment (ddgenome) [inline index.ts]: Not relevant, but still nice.

File: lib/pack/goal-state/cancelGoals.ts [hunk 2]
```diff
@@ -204,11 +203,11 @@ async function cancelGoalSet(goalSetId: string, ctx: HandlerContext, id?: string
 
     if (result && result.SdmGoalSet && result.SdmGoalSet.length === 1) {
         const gs = result.SdmGoalSet[0];
-        const newGoalSet = {
+        const newGoalSet: any = {
```
Comment (ddgenome) [inline lib/pack/goal-state/cancelGoals.ts]: Is there not a type for goal set?

File: lib/pack/goal-state/manageGoalSets.ts [hunk 3]
```diff
@@ -115,13 +114,13 @@ export async function manageGoalSets(sdm: SoftwareDeliveryMachine,
         const state = goalSetState(goals || []);
 
         if (state !== goalSet.state) {
-            const newGoalSet = {
+            const newGoalSet: any = {
```
Comment (ddgenome) [inline lib/pack/goal-state/manageGoalSets.ts]: Another goal set type?

File: package.json [hunk 4]
```diff
@@ -61,9 +61,9 @@
     "@atomist/tree-path": "^1.0.3"
   },
   "devDependencies": {
-    "@atomist/automation-client": "2.0.0-master.20191207145203",
+    "@atomist/automation-client": "2.0.0-master.20191213160252",
     "@atomist/microgrammar": "^1.2.1",
-    "@atomist/sdm": "2.0.0-master.20191207100157",
+    "@atomist/sdm": "2.0.0-sdm-mutations.20191213145220",
```
Comment (ddgenome) [inline package.json]: Do you want to wait until the sdm PR is merged and this updated to merge this?

File: lib/pack/goal-state/cancelGoals.ts (hunk 2 above)
Comment (cdupuis) [reply lib/pack/goal-state/cancelGoals.ts]: Those graphql query types and the SdmGoalSetMessage aren’t compatible. The query types are all optional. 😞

File: lib/pack/goal-state/manageGoalSets.ts (hunk 3 above)
Comment (cdupuis) [reply lib/pack/goal-state/manageGoalSets.ts]: See above. Not sure what to do.

File: package.json (hunk 4 above)
Comment (cdupuis) [reply package.json]: Just did that. Thanks.

**Pull request merged**